iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs

```diff
@@ -710,6 +710,14 @@ private async void ProcessMessage(IChannelHandlerContext context, Packet packet)
 
                     case PacketType.PUBLISH:
                         ProcessPublish(context, (PublishPacket)packet);
+                        // Checking if acknowledgement is needed after receiving direct method from Hub with QoS = 1,
+                        // otherwise its not acknowledged because direct method response does not require CompleteAsync() call.
+                        // Other messages with QoS = 1 are acknowledged from WriteAsync() after CompleteAsync() call.
+                        if (((PublishPacket)packet).QualityOfService == QualityOfService.AtLeastOnce &&
+                            ((PublishPacket)packet).TopicName.StartsWith(MqttTransportHandler.MethodPostTopicPrefix, StringComparison.OrdinalIgnoreCase))
+                        {
+                            await AcknowledgeAsync(context, ((PublishPacket)packet).PacketId.ToString(CultureInfo.InvariantCulture)).ConfigureAwait(false);
```

Comment:
ToString method required IFormatProvider to be specified, so added CultureInfo.InvariantCulture